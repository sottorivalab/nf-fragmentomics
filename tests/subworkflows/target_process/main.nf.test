nextflow_workflow {

    name "Test Workflow TARGET_PROCESS"
    script "subworkflows/target_process/main.nf"
    workflow "TARGET_PROCESS"

    test("Should run without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            workflow {
                """
                input[0] = Channel.of([
                    'test_source', ['test'], [file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/bed/test.bed', checkIfExists: true)]
                ])
                input[1] = Channel.of([
                    [ caseid:'test', sampleid: 'test', timepoint: 'T1' ],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina//bigwig/test.bigwig', checkIfExists: true)
                ])
                input[2] = [file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.blacklist_intervals.bed', checkIfExists: true)]
                """
            }
        }

        then {
            assert workflow.success            
        }

    }

}
