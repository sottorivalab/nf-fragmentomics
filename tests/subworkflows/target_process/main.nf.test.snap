{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "caseid": "test",
                            "sampleid": "test",
                            "timepoint": "T1"
                        },
                        {
                            "source": "test",
                            "name": "test"
                        },
                        "test_test_test_peak_data.tsv:md5,7582b17bccfd8dd5e51b5a7910a08a38",
                        "test_test_test_peak_stats.tsv:md5,405d1d60d8b80c4c55eaef09793977f2",
                        "test_test_test_RawSignal.pdf:md5,bf446def35fab2b1eab3d21da31de741",
                        "test_test_test_RelativeSignal.pdf:md5,c952403cdfece89cfddee791885a454c"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-22T15:35:08.067795"
    }
}